lib/facility.rb

```diff
@@ -1,14 +1,81 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees
+              
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(info_hash, services = [])
+    @name = info_hash[:name]
+    @address = info_hash[:address]
+    @phone = info_hash[:phone]
+    @services = services
```

Comment:
This is indeed a possible solution for handling services! Nice work! You could also just keep the one argument as info_hash, and in this initialize here  with a conditional:
```
if info_hash.has_key?(:services)
   @services = info_hash[:services]
else
   @services = []
end
```